### PR TITLE
chore: profile page changes

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/utils/LogicUtils.res [hunk 1]
```diff
@@ -342,6 +342,10 @@ let camelToSnake = str => {
   ->String.toLowerCase
 }
 
+let userNameToTitle = str => {
+  str->String.split(".")->Array.map(x => capitalizeString(x))->Array.joinWith(" ")
+}
+
```
Comment (PritishBudhiraja) [inline src/utils/LogicUtils.res:348]: Single line remove {}
and 
```
let userNameToTitle = str => str->String.split(".")->Array.map(capitalizeString)->Array.joinWith(" ")
```

File: src/screens/Settings/HSwitchProfile/HSwitchProfileSettingsEntity.res [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    is_active: getBool(dict, "is_active", false),
+    role_id: getString(dict, "role_id", ""),
+    role_name: getString(dict, "role_name", ""),
+    org_id: getString(dict, "org_id", ""),
+  }
+}
+
+let getItems: JSON.t => array<switchMerchantListResponse> = json => {
+  LogicUtils.getArrayDataFromJson(json, itemToObjMapper)
+}
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/HSwitchProfileSettingsEntity.res:27]: explicit type not needed. Just write the func

File: src/screens/Settings/HSwitchProfile/HSwitchProfileSettingsEntity.res [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+let getHeading = colType => {
+  switch colType {
+  | MerchantName => Table.makeHeaderInfo(~key="merchant_name", ~title="Merchant Name", ())
+  | RoleName => Table.makeHeaderInfo(~key="role", ~title="Role", ())
+  }
+}
+
+let getCell = (item: switchMerchantListResponse, colType): Table.cell => {
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/HSwitchProfileSettingsEntity.res:36]: can we remove the Table.cell not needed

File: src/screens/Settings/HSwitchProfile/HSwitchProfileSettingsEntity.res [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      <div className="flex flex-row gap-2">
+        <Icon name="user" className="text-jp-gray-700" size=12 />
+        <span> {item.role_name->LogicUtils.snakeToTitle->React.string} </span>
+      </div>,
+      "",
+    )
+  }
+}
+
+let merchantTableEntity = () =>
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/HSwitchProfileSettingsEntity.res:50]: Can we remvove this function call as we are not passing any parameter just use the entity value

File: src/screens/Settings/HSwitchProfile/HSwitchProfileSettings.res [hunk 5]
```diff
@@ -1,56 +1,45 @@
 module MerchantDetailsSection = {
   @react.component
   let make = () => {
-    let fetchDetails = APIUtils.useGetMethod()
-    let (merchantInfo, setMerchantInfo) = React.useState(_ => Dict.make())
+    open HSwitchProfileSettingsEntity
     let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Success)
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/HSwitchProfileSettings.res:5]: Why default is Success?
Can you change this to Loading and on line 15 we can remove that.

File: src/screens/Settings/HSwitchProfile/HSwitchProfileSettings.res [hunk 6]
```diff
@@ -1,56 +1,45 @@
...
       } catch {
       | Exn.Error(_) => setScreenState(_ => PageLoaderWrapper.Custom)
       }
-    }
-    React.useEffect0(() => {
-      getMerchantDetails()->ignore
       None
     })
 
+    let (offset, setOffset) = React.useState(_ => 0)
```
Comment (PritishBudhiraja) [inline src/screens/Settings/HSwitchProfile/HSwitchProfileSettings.res:24]: Define in the top

**Review by PritishBudhiraja — DISMISSED** [review]

File: src/components/Table.res [hunk 7]
```diff
@@ -360,7 +362,7 @@ module TableHeadingCell = {
     let headerBgColor =
       headerCustomBgColor->Option.isSome
         ? headerCustomBgColor->Option.getOr("")
-        : "bg-gray-50 dark:bg-jp-gray-darkgray_background"
+        : "bg-offset_white dark:bg-jp-gray-darkgray_background"
```
[the remark this answers is not in the document]
Comment (gitanjli525) [reply src/components/Table.res:365]: yeah, that's the background color of header in table, won't be a problem

**Review by PritishBudhiraja — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**